students/hyelin/westargram/users/views.py

```diff
@@ -1,3 +1,95 @@
+from email import message
+import json 
+import re
+from urllib import response 
+##View
+from django.views import View
+##Model
+from .models import User
+##Http
+from django.http import JsonResponse
 
-# Create your views here.
+
+class SignUpView (View):
+    def post(self, request):
+
+        """
+        1. 목적 : 클라이언트로부터 받은 사용자 데이터
+        (이름, 이메일, 비밀번호, 연락처, 그외 개인정보)를 데이터베이스에 등록하기 위함
+        2. INPUT : 
+        {
+            "name"         : "hyelin",
+            "email"        : "rlafls9596@gamil.com",
+            "password"     : "rlagPFls95@",
+            "phone_number" : "01049093501"
+        }
+        3. OUTPUT : 
+        성공 시
+        {
+            "message" : "SUCCESS" , status=201
+        }
+        """
+        user_data   = json.loads(request.body)
+        REX_EMAIL   = r'^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+        REX_PASS    = r'^(?=.*[A-Za-z])(?=.*\d)(?=.*[$@$!%*#?&])[A-Za-z\d$@$!%*#?&]{8,}$'
+        
+        ## 1. name누락 
+        if 'name' not in user_data :
+            return JsonResponse({'message' : '이름을 입력해 주세요'}, status=400)
+        ## 2. email 누락 
+        if 'email' not in user_data :
+            return JsonResponse({'message' : '이메일을 입력해 주세요'}, status=400)
+        ## 3. password 누락 
+        if 'password' not in user_data :
+            return JsonResponse({'message': '비밀번호를 입력해 주세요'}, status=400)
+        ## 4. 이메일 양식 비충족
+        if not re.fullmatch(REX_EMAIL, user_data['email']) :
+            return JsonResponse({'message' : '유효하지 않은 이메일입니다'})
+        ## 5. password 양식 비충족 
+        if not re.fullmatch(REX_PASS, user_data['password']) :
+            return JsonResponse({'message' : '잘못된 형식의 비밀번호 입니다'}, status=401)
+        ## 6. 이메일 중복시 -> 이미 가입된 email입니다
+        if not User.objects.filter(email = user_data['email']).exists() :
+            User.objects.create(
+                name = user_data['name'],
+                password = user_data['password'],
+                email = user_data['email'],
+                phone = user_data['phone']  
+            )
+            return JsonResponse({'message' : 'SUCCESS'}, status=201)
+        
+        return JsonResponse({'message' : '이미 가입된 회원 입니다'}, status=401)
+
+class LoginView(View):
+    def post(self, request):
+
+        """
+        1. 목적 : 클라이언트로부터 받은 로그인 목적의 사용자 데이터
+        (이메일과 비밀번호를 데이터베이스에 조회하여 일치하는지 확인하고  
+        일치 여부에 따라 로그인 성공/실패를 결정)
+        2. INPUT : 
+        {
+            "email"        : "rlafls9596@gamil.com",
+            "password"     : "rlagPFls95@",
+        }
+        3. OUTPUT : 
+        성공 시
+        {
+            "message" : "SUCCESS" , status=201
+        }
+        """
+
+        data = json.loads(request.body)
+
+        if 'email' not in data  or 'password' not in data :
+            return JsonResponse( {"message": "KEY_ERROR"}, status=400)
+            
+        if User.objects.filter(email=data['email']).exists() :
+            login_info = User.objects.get(email__exact=data['email'])
```

Comment:
`User.objects.get(email__exact=data['email'])` 코드가 실행되면 반환하는 값은 로그인 정보가 아니라 사용자(user) 정보에요!
변수명을 정할 때 변수가 참조하고 있는 객체가 어떤 값을 담고 있는지 알 수 있도록 만들어주세요~

```suggestion
            user = User.objects.get(email__exact=data['email'])
```

Reply:
- [x]

Reply:
✅